swap/src/bitcoin/transactions.rs

```diff
@@ -14,7 +14,11 @@ use std::collections::HashMap;
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct TxLock {
-    inner: Transaction,
+    // todo: We only need to send the transaction here but bdk_wallet.sign() wants a psbt.
+    // Constructing the psbt from the tx was failing after the introduction of psbt so we send the
+    // full psbt over. This is not ideal and has resulted in several clones to get the txid in the
+    // rest of this file.
+    inner: PartiallySignedTransaction,
```

Comment:
Using PSBT internally is for the best. I don't see the clone as an issue. I'd recommend to remove this TODO.